infra/modules/azure_function_app/function_app_slot.tf

```diff
@@ -46,8 +46,15 @@ resource "azurerm_linux_function_app_slot" "this" {
       FUNCTIONS_WORKER_PROCESS_COUNT = local.function_app.worker_process_count,
     },
     local.application_insights.enable ? {
+      # SDK AI Sampling (for dependencies)
       # https://docs.microsoft.com/en-us/azure/azure-monitor/app/sampling
-      APPINSIGHTS_SAMPLING_PERCENTAGE = "100"
+      APPINSIGHTS_SAMPLING_PERCENTAGE = "100",
```

Comment:
@Krusty93 we should let this be configurable as well (same defaults as production?)

Reply:
Is this because of the canary deployment?

Reply:
Dunno. I guess it won't be affected anyway.

Reply:
For the canary deployment we what that all the API call must be logged, a single failure in a single endpoint with a very low request rate is crucial.
Es. the service has 2 API, the first with a call rate of 1k req/min the second with 1 req/min. The sampling will make the canary monitoring on the second inefficient.